Assets/Scripts/Hazards.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class Hazards : MonoBehaviour {
+
+    private bool isOn = true; //Used to determine whether or not the hazard is currently showing
+    private float timeToChange = .015f; //Used to determine the time between hazard changing show state
+    private float timeToAdd = 0.015f; //Add to timeToChange to continually change the hazards renderer
+    [SerializeField]
+    Buttons button; //Button associated with this gate
+    [SerializeField]
+    SoundManager sound;
+
+    private void Update()
+    {
```

Comment:
Same thing as Update for PlayerController, you need to divide up all these actions into functions and shifting them around to further down in the code so Update isn't so crowded and anyone looking at it can know what it's at least supposed to be doing from a short glance